### PR TITLE
feat: add affinity and node affinity as values 

### LINKED ISSUE #567: Chart Enhancement - tolerations, nodeSelector, resources
**Is your feature request related to a problem?/Why is this needed**

Please add options to configure tolerations, nodeSelector and resources blocks for pods launched by this chart. 

**Describe the solution you'd like in detail**

Standard public chart value options for customizing/configuring workloads in various cluster configurations.

See `helm create` for the standard helm chart template.

could refer to https://github.com/helm/charts/tree/master/stable/cert-manager

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by cyril-corbon**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
- [x] add node affinity as values in the helm chart for daemonset
- [x] add affinity to controller deployment

**Which issue(s) this PR fixes**:

Fixes #567

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
could you only make change in https://github.com/kubernetes-sigs/azurefile-csi-driver/tree/master/charts/latest?

File: charts/v1.6.0/azurefile-csi-driver/values.yaml [hunk 1]
```diff
@@ -60,6 +60,14 @@ controller:
     - key: "node-role.kubernetes.io/controlplane"
       operator: "Exists"
       effect: "NoSchedule"
+  # nodeAffinity:
```
Comment (andyzhangx) [inline charts/v1.6.0/azurefile-csi-driver/values.yaml:63]: don't comment here

File: charts/v1.6.0/azurefile-csi-driver/templates/csi-snapshot-controller.yaml [hunk 2]
```diff
@@ -22,6 +22,11 @@ spec:
 {{- with .Values.controller.tolerations }}
       tolerations:
 {{ toYaml . | indent 8 }}
+{{- end }}
+{{- with .Values.controller.nodeAffinity }}
+      affinity:
+        nodeAffinity:
+{{ toYaml . | indent 10 }}
```
Comment (cyril-corbon) [inline charts/v1.6.0/azurefile-csi-driver/templates/csi-snapshot-controller.yaml:29]: Maybe we could change `.Values.controller.nodeAffinity` to `.Values.snapshot.snapshotController.affinity`.

we may want to split snapshot controller affinity and azurefile controller :thinking:  ?

**Comment by cyril-corbon** [comment]
I moved the files to charts/latest and I modify the controller `nodeAffinity` to `affinity`

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

**Review by Sissi44 — APPROVED** [review]

**Comment by andyzhangx** [comment]
latest chart config has changed, pls run "helm package charts/latest/azurefile-csi-driver -d charts/latest/" to update tgz file

**Comment by cyril-corbon** [comment]
I don't think these tests meaningful 🤔,  can we merge it ?

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Comment by andyzhangx** [comment]
> I don't think these tests meaningful 🤔, can we merge it ?

@cyril-corbon these tests are crucial, and test failures are fixed. We just overcome a crucial test failure which happened in most k8s tests in k8s community, finally recovered after one week.

**Comment by cyril-corbon** [comment]
> > I don't think these tests meaningful 🤔, can we merge it ?
> 
> @cyril-corbon these tests are crucial, and test failures are fixed. We just overcome a crucial test failure which happened in most k8s tests in k8s community, finally recovered after one week.

I wanted to said that the test result was not linked to this PR, (meaningful was not the right terms). Well done for the fix 😁 and thanks for the review / the merge.

**Pull request merged**